app/screens/settings-screen/settings-screen.tsx

```diff
@@ -162,6 +163,15 @@ export const SettingsScreenJSX: React.FC<SettingsScreenProps> = (params) => {
       enabled: !isAuthed,
       greyed: isAuthed,
     },
+    {
+      category: LL.common.username(),
+      icon: "person",
+      id: "username",
+      subTitleDefaultValue: LL.SettingsScreen.tapUserName(),
+      subTitleText: username,
```

Comment:
this will be undefined if the `username` is not set. instead we want the button to still be enabled and send to the set username . I think all the logic should be in `addressScreen`

Reply:
do i need to make a screen just like addressScreen in which i can edit my username  or just need to navigate to adressScreen on clicking username

Reply:
hey i added these line
   action: () => navigation.navigate("addressScreen"),
      enabled: isAuthed && username === undefined,
      greyed: !isAuthed || username !== undefined,